src/_pytest/faulthandler.py

```diff
@@ -24,7 +25,9 @@ def pytest_addoption(parser: Parser) -> None:
 def pytest_configure(config: Config) -> None:
     import faulthandler
 
-    config.stash[fault_handler_stderr_fd_key] = os.dup(get_stderr_fileno())
+    stderr_fileno = get_stderr_fileno()
```

Comment:
We need to document that we cannot restore the original fd due to lack of complete enough api in the stdlib